crates/network-scheduler/src/assignment.rs

```diff
@@ -23,11 +29,19 @@ pub struct Dataset {
 
 #[derive(Debug, Default, Serialize, Deserialize)]
 #[serde(rename_all = "kebab-case")]
-struct EncryptedHeaders {
+struct Headers {
     worker_id: String,
     worker_signature: String,
 }
 
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+struct EncryptedHeaders {
+    identity: Vec<u8>,
+    nonce: Vec<u8>,
+    ciphertext: Vec<u8>,
```

Comment:
Can we make it a map of `name`: `encrypted_value` instead and encrypt both with the same key and nonce? Parsing with serde_json is very expensive, so it's better to avoid it where possible.

Reply:
Nonce have to be unique, but I can pull key up